### PR TITLE
Github Action to label PR

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to replace astropy-bot labeler with GitHub Action.

The labeling isn't as programmatic as astropy-bot but it is only a one-time cost to set up and in the end more flexible and easier to maintain than the astropy-bot version.

**TODO**

- [x] Fix labeling YAML so labels apply correctly.
- [x] Maybe we don't need to apply `testing` label every time `tests` is touched? For example, PR that fixes a bug and adds a test to prevent regression usually doesn't have `testing` label applied to it.

(Skipped CI for now.)

### CODE REVIEW

**Comment by saimn** [comment]
Are you sure you need to duplicate `somedir/*` and `somedir/**/*` ? I think the latter should be enough, but it's not so clear from the README.

**Comment by pllim** [comment]
@saimn , not sure. I am following the README example. If there is better doc somewhere else, would be nice... 😬

File: .github/workflows/label_pr.yml
Comment (saimn) [inline .github/workflows/label_pr.yml:11]:
```suggestion
    - uses: actions/labeler@main
```
It seems they stopped tagging versions and many features are in ~~master~~main

**Comment by saimn** [comment]
Yeah, the README is confusing. The first example suggests that it is not needed
```
# Add 'label1' to any changes within 'example' folder or any subfolders
label1:
  - example/**/*
```
but then there is another example with both `*` and `**/*` :man_shrugging:

File: .github/workflows/label_pr.yml
Comment (pllim) [reply .github/workflows/label_pr.yml:11]: I was hoping I don't have to live dangerously with `main` but I guess I have no choice. Thanks for pointing that out!

**Comment by pllim** [comment]
So... It kinda works but not correctly at the current state.

Case study: kali-maa/astropy#5
Expected: `Docs` `testing` `wcs.wcsapi`
Got: `Docs` `testing` `wcs.wcsapi` `utils` `visualization` (???)

**Comment by pllim** [comment]
Also, it appears that the Action automatically creates a label for you if you tell it to use some label that doesn't exist yet.

File: .github/labeler.yml
Comment (astrofrog) [inline .github/labeler.yml:56]: Can we combine these three entries? At least we could do ``*/constants/*`` for the first right? (then don't need docs one)

Comment (saimn) [reply .github/labeler.yml:56]: Yes, after doing some testing `'**/constants/**/*'` seems to work.

**Comment by pllim** [comment]
For some reason, `astropy-bot` hates me now. It refused to update its judgement after I removed the "Experimental" label. This is ready for review. Thanks to much help from @saimn , we got the logic sorted.

You can see it in action (pun?) over at https://github.com/kali-maa/astropy/pulls . You can also test it out yourselves by opening PRs against `master` branch of that fork.

File: pyproject.toml
Comment (pllim) [inline pyproject.toml:13]: This depends on astropy/astropy-bot#123 being merged and deployed.

File: pyproject.toml
Comment (bsipocz) [inline pyproject.toml:13]: I think the comment out way would only work if the default is `False`. My problem was in the other PR, is that if the default is True, and there is only string bool eval, then the `'false'` below would be a `True`, and thus there is no way to disable from the default.

Comment (pllim) [reply pyproject.toml:13]: I am not familiar with TOML parsing. I thought a "false" would be parsed as `False` (boolean)? I am following some other existing code. I am confused now.

Comment (bsipocz) [reply pyproject.toml:13]: ah, ok, that parsing is a possibility. Either case, as you said it all followed the existing code (for which I didn't see much testing, thus couldn't double checked it, but that's an issue for the bot/baldrick).

**Comment by bsipocz** [comment]
@pllim - I've merged the other PR. Could you redeploy the bot?

File: pyproject.toml
Comment (pllim) [reply pyproject.toml:13]: For future reference: https://github.com/astropy/astropy-bot/pull/123#issuecomment-723130381 (I think the config is correct.)

**Comment by pllim** [comment]
The bot was deployed on merge of astropy/astropy-bot#123. 😄 🤞

**Comment by embray** [comment]
Wow, you're so ahead of the game.  I was about to look into doing something just like this and it looks like I'm already more than a week late.  This is a problem with having only a part-time gig...  Instead I will read and review this to learn from it.

File: .github/labeler.yml
Comment (embray) [inline .github/labeler.yml:19]: Why the single-quotes on this one?

Comment (embray) [reply .github/labeler.yml:19]: Ditto on a few other examples below.  It should be harmless but it's not clear to me that they're needed either.

**Comment by astrofrog** [comment]
It would be cool to extend this one day to include auto assigning reviewers!

**Comment by bsipocz** [comment]
I'm pretty sure I saw that either as a bot or an action somewhere in action.

**Comment by embray** [comment]
> It would be cool to extend this one day to include auto assigning reviewers!

I've been thinking of doing that, in combination with our updated team database (e.g. https://github.com/astropy/astropy-project/issues/136#issuecomment-706095664); it might require some customization over any existing auto-assign action, but it could be used to associate people in specific roles with GH labels, for example.

File: .github/labeler.yml
Comment (saimn) [reply .github/labeler.yml:19]: It throws a YAML exception without the quotes.

Comment (embray) [reply .github/labeler.yml:19]: Oh, of course, because of the `*`!  Thank you.

**Comment by pllim** [comment]
> auto assigning reviewers

That is already possible without Actions, see https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners . We use it (in a very limited way) over at https://github.com/spacetelescope/notebooks/blob/master/CODEOWNERS . I believe that we discussed this a long time ago and there were concerns that implementing this would overburden some while alienating others, so we never did it.

**Comment by pllim** [comment]
So, back to just labeling PRs, should we merge this and try it out? Or do you want to play with it a bit more before merge?

**Review by astrofrog — APPROVED** [review]
Looks good, thanks! I think we can just merge and refine later if needed 👍

**Comment by pllim** [comment]
🤞 😬 🤞

**Comment by embray** [comment]
Regarding auto-assigning maintainers I think it should be implemented but on an opt-in basis by said maintainers.  I.e. they would have to explicitly agree to actually be assigned such issues.  That said, if they aren't able to respond over a very long period of time maybe they should receive "emeritus" status :)

**Comment by pllim** [comment]
Let's move the auto-assign maintainers discussions to astropy/astropy-bot#101 ?

**Pull request merged**
